### PR TITLE
Consider neuroglancer transform half a voxel shift when fetching the affine

### LINKED ISSUE #622: 0.5 voxel affine shift of neuroglrancer/precomputed sources
siibra-python (v1.0a19 an prior) fetches the affine neuroglrancer/precomputed images from `transform.json` files stored along with the images (`baseurl/transform.json`). However, these transform files are only meant for neuroglancer viewer to begin with. Therefore, it is `NeuroglrancerProvider`'s job to correctly fetch the affine correctly from the source. Will be remedied with 1.0a20.

Before closing update:
- EDIT: https://github.com/FZJ-INM1-BDA/siibra-configurations/issues/77
- [x] update tests concerning `BigBrainIntesnsityProfile`s

### PR DESCRIPTION
**Pull request by AhmetNSimsek**
fixes #622

### CODE REVIEW

**Comment by AhmetNSimsek** [comment]
@xgui3783 This should be applied to meshes and surface meshes as well, right?

File: siibra/volumes/providers/neuroglancer.py [hunk 1]
```diff
@@ -37,6 +37,32 @@
 import json
 
 
+def get_affine_from_ng_transfrom(transform_nm: np.ndarray, scaling: np.ndarray) -> np.ndarray:
+    """
+    transfrorm.json stored with neuroglancer precomputed images and meshes
+    are meant to be used for neuroglancer viewers and hence they are not
+    representative of the affine in other tools. This method convertes them
+    into a standard affine, that is, converts to mm from nm and shifts back
+    0.5mm in each axis.
```
[the remark this answers is not in the document]
Comment (AhmetNSimsek) [reply siibra/volumes/providers/neuroglancer.py:46]: Oh yes! I forgot to push my latest changes, my bad 😓

**Comment by AhmetNSimsek** [comment]
> > @xgui3783 This should be applied to meshes and surface meshes as well, right?
> 
> negative.
> 
> meshes are unaffected. only images are affected.

It turns out, siibra-python still uses the transform.json stored at the same url. Therefore, the affine needs to be updated accordingly.

File: siibra/volumes/providers/neuroglancer.py [hunk 2]
```diff
@@ -37,6 +37,32 @@
...
+    representative of the affine in other tools. This method shifts back
+    half a voxel in each axis.
+
+    Parameters
+    ----------
+    transform_nm: np.ndarray
+        Transform array created for dispalying an image correctly from
+        neuroglancer precomputed format in neuroglancer viewer.
+    resolution_nm: np.ndarray
+        The voxel resolution in nanometers.
```
Comment (xgui3783) [inline siibra/volumes/providers/neuroglancer.py:53]: voxel resolution of the **highest** level of resolution

**Review by xgui3783 — APPROVED** [review]
Looks good. 

I would update the docstring, but maybe also add a note in doc string, link to https://neuroglancer-scripts.readthedocs.io/en/latest/neuroglancer-info.html#different-conventions-for-coordinate-transformations

**Pull request merged**